source/docs/software/wpilib-overview/using-test-mode.rst

```diff
@@ -0,0 +1,71 @@
+Using test mode
+===============
+
+Test mode is designed to enable programmers to have a place to put code to verify that all systems on the robot are functioning. In each of the robot program templates there is a place to add test code to the robot. If you use the TimedRobot template, or use command-based programming you can easily see the operation of all the motors and sensors through the LiveWindow.
+
+Enabling Test mode
+------------------
+
+Test mode on the robot can be enabled from the Driver Station just like Autonomous or Teleop. When in test mode, the testInit method is run once, and the testPeriodic method is run once per tick, in addition to RobotPeriodic, the same as Teleop and Autonomous control modes. To enable test mode in the Driver Station, select the "Test" button and enable the robot. The test mode code will then run.
+
+Adding Test mode code to your robot code
+----------------------------------------
+
+Adding test mode can be as painless as calling your already written Teleop methods from Test. This will allow you to use the LiveWindow tuning features for classes such as PIDBase and PIDController, as well as PIDSubsystems and PIDCommands, to change PIDF constants and setpoints. Make sure to add your subsystems to SmartDashboard with the ``putData(subsystem)`` and ``PutData(subsystem)`` methods.
```

Comment:
```suggestion
Adding test mode can be as painless as calling your already written Teleop methods from Test. This will allow you to use the LiveWindow tuning features for classes such as PIDBase and PIDController, as well as PIDSubsystems and PIDCommands, to change PID constants and setpoints. Make sure to add your subsystems to SmartDashboard with the ``putData(subsystem)`` or ``PutData(subsystem)`` method.
```